gpflow/likelihoods/likelihoods.py

```diff
@@ -391,16 +391,31 @@ def integrand(*X):
 
 class Gaussian(ScalarLikelihood):
     r"""
-    The Gaussian likelihood is appropriate where uncertainties associated with the data are
-    believed to follow a normal distribution, with constant variance.
+    The Gaussian likelihood is appropriate where uncertainties associated with
+    the data are believed to follow a normal distribution, with constant
+    variance.
 
     Very small uncertainties can lead to numerical instability during the
-    optimization process. A lower bound of 1e-6 is therefore imposed on the likelihood variance
-    by default.
+    optimization process. A lower bound of 1e-6 is therefore imposed on the
+    likelihood variance by default.
     """
 
-    def __init__(self, variance=1.0, variance_lower_bound=1e-6, **kwargs):
+    DEFAULT_VARIANCE_LOWER_BOUND = 1e-6
+
+    def __init__(self, variance=1.0, variance_lower_bound=DEFAULT_VARIANCE_LOWER_BOUND, **kwargs):
```

Comment:
This is error prone approach. Changing `DEFAULT_VARIANCE_LOWER_BOUND` in the class doesn't change the default value of `variance_lower_bound` argument. Therefore, I would advise you to move this assignment to the init with `if else` statement, whereas the default value would be `None`.

Reply:
I added this following a (very valuable) suggestion in #1407, to allows downstream code to check what the default value *is*. It's not meant to be changed, which is why it's all-caps (the python convention for constants that you should not go around and change).

Reply:
I think it might be more confusing for `None` to mean "default value" rather than "no lower bound" :thinking: